### PR TITLE
ref(ui): Convert assignee selector to class component

### PR DESCRIPTION
**Pull request by scttcper**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/dropdownAutoComplete/menu.tsx [hunk 1]
```diff
@@ -22,7 +22,8 @@ export type MenuFooterChildProps = {
 type ListProps = React.ComponentProps<typeof List>;
 
 type Props = {
-  items: ItemsBeforeFilter;
+  /** null items indicates loading */
+  items: ItemsBeforeFilter | null;
```
Comment (scttcper) [inline src/sentry/static/sentry/app/components/dropdownAutoComplete/menu.tsx:26]: must allow null see itemsLoading
```
const itemsLoading = items === null
```

👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺👺🧨

File: src/sentry/static/sentry/app/components/assigneeSelector.tsx [hunk 2]
```diff
@@ -129,58 +119,75 @@ const AssigneeSelectorComponent = createReactClass<Props, State>({
...
-  assignableTeams(): AssignableTeam[] {
-    if (!this.props.id) {
-      return [];
+  onGroupChange(itemIds: Set<string>) {
+    if (!itemIds.has(this.props.id)) {
+      return;
     }
+    const group = GroupStore.get(this.props.id);
+    this.setState({
+      assignedTo: group && group.assignedTo,
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/assigneeSelector.tsx:153]:
```suggestion
      assignedTo: group?.assignedTo,
```

File: src/sentry/static/sentry/app/components/assigneeSelector.tsx [hunk 3]
```diff
@@ -129,58 +119,75 @@ const AssigneeSelectorComponent = createReactClass<Props, State>({
...
+      suggestedOwners: group && group.owners,
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/assigneeSelector.tsx:154]:
```suggestion
      suggestedOwners: group?.owners,
```

Could these use optional chaining?

File: src/sentry/static/sentry/app/components/assigneeSelector.tsx [hunk 4]
```diff
@@ -25,18 +24,23 @@ import GroupStore from 'app/stores/groupStore';
...
+type AssignableTeam = {
+  id: string;
+  display: string;
+  email: string;
+  team: Team;
+};
+
 type Props = {
-  id: string | null;
+  id: string;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/assigneeSelector.tsx:43]: are you sure about removing the `null` type? I noticed that the defaultProps still set it to `null`

https://github.com/getsentry/sentry/blob/a0738f5575126a4c05bdf0f54819a39b11912509/src/sentry/static/sentry/app/components/assigneeSelector.tsx#L67

maybe we could make it optional and remove it from the defaultProps object?

Comment (scttcper) [reply src/sentry/static/sentry/app/components/assigneeSelector.tsx:43]: Removed the default prop, everywhere that uses it seems to be typed as string, and all the places we use it don't allow for it to be null.

**Review by markstory — APPROVED** [review]

**Pull request merged**